packages/mdc-checkbox/mdc-checkbox.scss

```diff
@@ -14,334 +14,137 @@
 // limitations under the License.
 //
 
-@import "@material/animation/functions";
-@import "@material/ripple/mixins";
-@import "@material/rtl/mixins";
+@import "@material/theme/mixins";
+@import "./mixins";
 @import "./variables";
-@import "./keyframes";
-
-@function mdc-checkbox-transition-enter($property, $delay: 0ms, $duration: $mdc-checkbox-transition-duration) {
-  @return mdc-animation-enter($property, $duration, $delay);
-}
-
-@function mdc-checkbox-transition-exit($property, $delay: 0ms, $duration: $mdc-checkbox-transition-duration) {
-  @return mdc-animation-exit-temporary($property, $duration, $delay);
-}
-
-@mixin mdc-checkbox-cover-element {
-  position: absolute;
-  top: 0;
-  right: 0;
-  bottom: 0;
-  left: 0;
-}
 
 // postcss-bem-linter: define checkbox
+
 .mdc-checkbox {
+  @include mdc-checkbox-base_;
   @include mdc-ripple-base;
-  @include mdc-ripple-bg((pseudo: "::before", theme-style: primary, opacity: .14));
-  @include mdc-ripple-fg((pseudo: "::after", theme-style: primary, opacity: .14));
-
-  display: inline-block;
-  position: relative;
-  flex: 0 0 $mdc-checkbox-size;
-  box-sizing: content-box;
-  width: $mdc-checkbox-size;
-  height: $mdc-checkbox-size;
-  padding: ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2;
-  line-height: 0;
-  white-space: nowrap;
-  cursor: pointer;
-  vertical-align: bottom;
+  @include mdc-checkbox-color(secondary);
 
   &::before,
   &::after {
-    top: 0;
-    left: 0;
-    width: 100%;
-    height: 100%;
-  }
-
-  // stylelint-disable plugin/selector-bem-pattern
-  &.mdc-ripple-upgraded--unbounded {
-    .mdc-checkbox__background::before {
-      content: none;
-    }
-  }
-  // stylelint-enable plugin/selector-bem-pattern
-
-  &__background {
-    @include mdc-checkbox-cover-element;
-    @include mdc-rtl-reflexive-position(
-      left, ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2, ".mdc-checkbox");
-
-    display: inline-flex;
-    top: ($mdc-checkbox-touch-area - $mdc-checkbox-size) / 2;
-    align-items: center;
-    justify-content: center;
-    box-sizing: border-box;
-    width: $mdc-checkbox-ui-pct;
-    height: $mdc-checkbox-ui-pct;
-    transition:
-      mdc-checkbox-transition-exit(background-color),
-      mdc-checkbox-transition-exit(border-color);
-    border: $mdc-checkbox-border-width solid $mdc-checkbox-border-color;
-    border-radius: 2px;
-    background-color: transparent;
-    pointer-events: none;
-    will-change: background-color, border-color;
-
-    @include mdc-theme-dark(".mdc-checkbox") {
-      border-color: $mdc-checkbox-border-color-dark;
-    }
-
-    // The frame's ::before element is used as a focus indicator for the checkbox
-    &::before {
-      @include mdc-checkbox-cover-element;
-      @include mdc-theme-prop(background, primary);
```

Comment:
FYI, #1364 is adding $edgeOptOut: true here, so you will want to do likewise wherever the equivalent is (not sure if that will require doing it in all invocations of a mixin now, which will generate more CSS...)